tests/js/client/aql/aql-optimizer-push-limit-into-index.js

```diff
@@ -0,0 +1,187 @@
+/*jshint globalstrict:false, strict:false, maxlen: 500 */
+/*global assertTrue, assertEqual, assertNotEqual, assertNull, assertFalse */
+
+// //////////////////////////////////////////////////////////////////////////////
+// / DISCLAIMER
+// /
+// / Copyright 2014-2024 ArangoDB GmbH, Cologne, Germany
+// / Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+// /
+// / Licensed under the Business Source License 1.1 (the "License");
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const internal = require("internal");
+const errors = internal.errors;
+const db = require("@arangodb").db;
+const helper = require("@arangodb/aql-helper");
+const assertQueryError = helper.assertQueryError;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function optimizerPushLimitIntoIndexTestSuite () {
+  let c;
+  return {
+    setUpAll : function () {
+      db._drop("UnitTestsCollection");
+      c = db._create("UnitTestsCollection"); 
+      var docs = []; 
+      for (i = 0; i < 1000; ++i) { 
```

Comment:
note: there should be no need to ever use JavaScript's `var` in new tests. `let` or `const` should be preferred, because they have clearer scoping.
```suggestion
      let docs = []; 
      for (let i = 0; i < 1000; ++i) { 
```